lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2494,14 +2495,34 @@ void ClangImporter::lookupValue(DeclName name, VisibleDeclConsumer &consumer){
     });
 }
 
-void
-ClangImporter::lookupTypeDecl(StringRef rawName, ClangTypeKind kind,
-                              llvm::function_ref<void(TypeDecl*)> receiver) {
+static Optional<ClangTypeKind>
+getClangTypeKindForNodeKind(Demangle::Node::Kind kind) {
```

Comment:
I really don't like moving this. Why is it the importer's responsibility to resolve manglings?

Reply:
This is basically based on a misunderstanding on my end that I since dicovered and I can revert this change. Because of how ClangTypeKind folds both enums and structs into ClangTypeKind::Tag I was afraid the we'd be loosing information that could make the type search in DWARF more accurate. Later I discovered that structs, unions and many enums are all mangled as structs anyway, so this point doesn't hold any more. I'll prepare a patch to undo this.

Reply:
```
enum class ClangTypeKind {
  Typedef,
  ObjCClass = Typedef,
  /// Structs, enums, and unions.
  Tag,
  ObjCProtocol,
}
```

Looks like we would be loosing the ability to distinguish between typedefs and ObjC classes, but that may not be such a big deal.